src/Stateless/MethodDescription.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace Stateless
+{
+    internal class MethodDescription
+    {
+        readonly string _description;                     // _description can be null if user didn't specify a description
+
+        public enum Timing
+        {
+            Synchronous,
+            Asynchronous
+        }
+        Timing _timing;
+
+        public static MethodDescription Create(Delegate method, string description, Timing timing = Timing.Synchronous)
+        {
+            return new MethodDescription(method?.TryGetMethodName(), description, timing);
+        }
+
+        private MethodDescription(string methodName, string description, Timing timing)      // description can be null if user didn't specify a description
+        {
+            MethodName = methodName;
+            _description = description;
+            _timing = timing;
+        }
+
+        /// <summary>
+        /// The name of the invoked method.  If the method is a lambda or delegate, the name will be a compiler-generated
+        /// name that is often not human-friendly (e.g. "(.ctor)b__2_0" except with angle brackets instead of parentheses)
+        /// </summary>
+        internal string MethodName { get; private set; }
```

Comment:
Thanks for the explanation. I don't think the rest of the code uses this convention, except perhaps in one or two places where we might be inconsistent. I'd rather we keep it consistently `public` here.